standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/tools/metatool/MetaToolTaskMetadataSummary.java

```diff
@@ -18,178 +18,348 @@
 
 package org.apache.hadoop.hive.metastore.tools.metatool;
 
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ArrayListMultimap;
 import com.google.gson.Gson;
 import com.google.gson.GsonBuilder;
-import org.apache.hadoop.hive.metastore.ObjectStore;
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
 
 import java.io.File;
 import java.io.IOException;
 import java.io.PrintWriter;
-import java.nio.file.Files;
-import java.nio.file.Path;
-import java.nio.file.Paths;
-import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.IdentityHashMap;
 import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.exception.ExceptionUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.common.TableName;
+import org.apache.hadoop.hive.metastore.api.MetaException;
+import org.apache.hadoop.hive.metastore.metasummary.MetaSummaryHandler;
+import org.apache.hadoop.hive.metastore.metasummary.MetaSummarySchema;
+import org.apache.hadoop.hive.metastore.metasummary.MetadataTableSummary;
+import org.apache.hadoop.hive.metastore.tools.MetaToolObjectStore;
+import org.apache.hadoop.hive.metastore.utils.JavaUtils;
+import org.apache.hadoop.hive.metastore.utils.MetaStoreUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class MetaToolTaskMetadataSummary extends MetaToolTask {
-    @Override
-    void execute() {
+  private static final Logger LOG = LoggerFactory.getLogger(MetaToolTaskMetadataSummary.class);
+  private static final Map<String, String> NON_NATIVE_SUMMARY_HANDLER = new HashMap<>();
+
+  static {
+    NON_NATIVE_SUMMARY_HANDLER.put("iceberg", "org.apache.iceberg.metasummary.IcebergSummaryHandler");
+  }
+
+  @Override
+  void execute() {
+    try {
+      String[] inputParams = getCl().getMetadataSummaryParams();
+      String formatOption = inputParams[0].toLowerCase().trim();
+      boolean formatJson = formatOption.equalsIgnoreCase("-json");
+      boolean formatCsv = formatOption.equalsIgnoreCase("-csv");
+      boolean formatConsole = formatOption.equalsIgnoreCase("-console");
+      if (!formatJson && !formatCsv && !formatConsole) {
+        System.err.println("Invalid option to -metadataSummary");
```

Comment:
Can we say in the error message that the format option is not provided?

Reply:
sure